scripts/game_structure/game_essentials.py

```diff
@@ -191,6 +193,46 @@ def update_game(self):
             self.switch_screens = True
         self.clicked = False
         self.keyspressed = []
+    
+    @staticmethod
+    def safe_save(path: str, write_data, max_attempts: int=15):
+        """ Attempt to safely safe a file.
+            If write_data is not a string, assumes you want this
+            in json format. 
+            Raises a RunTimeError is 
+            the file is nullied too many times. """
+        
+        # If write_data is not a string, 
+        if type(write_data) is not str:
+            _data = ujson.dumps(write_data, indent=4)
+        else:
+            _data = write_data
+            
+        dir_name, file_name = os.path.split(path)
+        
+        temp_file_path = get_temp_dir() + "/" + file_name + ".tmp"
+        i = 0
+        while True:
```

Comment:
The while loop is to save the file again if something goes wrong. Since file nullification seems to occur to random files, there is a chance that trying again will resolve the issue. 
If you'd prefer, I can refactor the while loop into a for loop.